sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py

```diff
@@ -86,3 +91,36 @@ def _create_pipeline(self, credential, **kwargs):
             DistributedTracingPolicy(),
         ]
         return config, AsyncPipeline(config.transport, policies=policies)
+
+    async def _batch_send(
+        self, *reqs: 'HttpRequest',
+        **kwargs
+    ):
+        """Given a series of request, do a Storage batch call.
+        """
+        request = self._client._client.post(  # pylint: disable=protected-access
+            url='https://{}/?comp=batch'.format(self.primary_hostname),
+            headers={
+                'x-ms-version': self._client._config.version  # pylint: disable=protected-access
+            }
+        )
+
+        request.set_multipart_mixed(
+            *reqs,
+            policies=[
+                StorageHeadersPolicy(),
+                self._credential_policy
+            ]
+        )
```

Comment:
Minor: There's some duplicate code between sync/async versions that could probably be moved into a separate sync function in the base class.

Reply:
That's right, but that's a global problem for this file: 80% of the code between `StorageAccountHostsMixin` and `AsyncStorageAccountHostsMixin` are duplication (just look at `_create_pipeline`...).
Because the two mixins don't share anything, they should both inherit from a `StorageAccountHostsMixinBase`, but I don't think I want to tackle in this very PR

Reply:
Agreed. Good call - save it for a post-release more extensive refactor :)

Reply:
Created an issue to track this separately https://github.com/Azure/azure-sdk-for-python/issues/7587